src/cloudwatch-mcp-server/awslabs/cloudwatch_mcp_server/cloudwatch_metrics/alarm_tools.py

```diff
@@ -0,0 +1,911 @@
+                    print(f"Composite Alarm: {alarm.alarm_name}")
+        """
+        try:
+            # Create region-specific client if region provided
+            cloudwatch_client = self._get_cloudwatch_client(region)
+            
+            # Get the actual region being used
+            # Handle case where region might be a Field object due to pydantic issues
+            if hasattr(region, 'default'):
+                actual_region = region.default or os.environ.get('AWS_REGION', 'us-east-1')
+            else:
+                actual_region = region or os.environ.get('AWS_REGION', 'us-east-1')
+            
+            # Fetch active alarms
+            logger.info(f'Fetching active alarms from region: {actual_region}')
+            
+            response = cloudwatch_client.describe_alarms(
+                StateValue='ALARM',
+                MaxRecords=50,
+                AlarmTypes=[
+                    'CompositeAlarm',
+                    'MetricAlarm'
+                ]
+            )
+            
+            # Transform results to summary models
+            metric_alarms = []
+            composite_alarms = []
+            
+            # Process metric alarms
+            for alarm in response.get('MetricAlarms', []):
+                metric_alarm = self._transform_metric_alarm(alarm)
+                metric_alarms.append(metric_alarm)
+            
+            # Process composite alarms
+            for alarm in response.get('CompositeAlarms', []):
+                composite_alarm = self._transform_composite_alarm(alarm)
+                composite_alarms.append(composite_alarm)
+            
+            total_count = len(metric_alarms) + len(composite_alarms)
+            has_more_results = 'NextToken' in response
+            
+            # Handle pagination if more than 50 alarms exist
+            if has_more_results:
+                logger.info(f'Found {total_count} alarms with more results available - handling pagination')
+                return await self._handle_pagination(ctx, cloudwatch_client, metric_alarms, composite_alarms, response.get('NextToken'), actual_region)
+            
+            # Handle empty results
+            message = None
+            if total_count == 0:
+                message = f"No active alarms found in region {actual_region}"
+            
+            logger.info(f'Found {total_count} active alarms ({len(metric_alarms)} metric, {len(composite_alarms)} composite)')
+            
+            return ActiveAlarmsResponse(
+                region=actual_region,
+                metric_alarms=metric_alarms,
+                composite_alarms=composite_alarms,
+                total_count=total_count,
+                has_more_results=has_more_results,
+                message=message
+            )
+            
+        except Exception as e:
+            logger.error(f'Error in get_active_alarms: {str(e)}')
+            return self._create_error_response(e)
```

Comment:
Do we need this wrapper error? Can't we simply throw the exception and let the agent deal with it (similarly to what we've done in the other tools)?